app/models/partner.rb

```diff
@@ -247,6 +247,47 @@ def has_service_areas?
     service_areas.any?
   end
 
+  def can_clear_address?(user = nil)
+    return false if address.blank?
+    return false if service_areas.empty?
+
+    return false if user.blank?
+    return true if user.root?
+    return true if user.admin_for_partner?(id)
+
+    return true if user.neighbourhood_admin?
+    return true if user.partnership_admin?
+  end
+
+  def warn_user_clear_address?(user)
+    return false if user.root?
+    return false if user.admin_for_partner?(id)
+
+    user_hood_ids = user.owned_neighbourhood_ids
+    # this shouldn't happen (?) - a user has no neighbourhoods
+    #   but can still remove a partners address? if that is the case
+    #   then they aren't seeing the partner by neighbourhood anyway
+    #   so warning them that removing the address will remove the
+    #   partner seems wrong
+    return true if user_hood_ids.empty?
```

Comment:
not possible to be able to edit a user without owning any neighbourhoods or meeting the first two criteria, so this can just be removed